### PR TITLE
daemon: add a Daemon argument to AccessChecker.CheckAccess

### PR DESCRIPTION
**Pull request by jhenstridge**
We want to make the theme installation REST API available to selected snaps over the `snapd-snap.socket` socket.  At present, the idea is to control which snaps have access to these calls through a marker interface. That means the access checker will need to be able to check whether a given snap has a connected plug of the marker interface.  I've done this by passing the `Daemon` as an argument to the`CheckAccess` method.

There are some alternatives that could be used instead:
* We really want the `Overlord` to perform the interface connection check, so we could pass that instead. There may be other future use cases that need more than the Overlord though.
* `Command` would also be usable here, since it holds a pointer to the `Daemon`.  It's not clear what other information in that struct could be useful to an access checker though.

### CODE REVIEW

**Comment by pedronis** [comment]
marking blocked until we clarify in which order we want to land the next batch of changes to access check and response related code

**Comment by jhenstridge** [comment]
I've updated the branch to take account of other recently merged changes.

**Review by pedronis — APPROVED** [review]
+1

**Review by mardy — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
